crates/c-api/include/wasmtime/linker.h

```diff
@@ -218,6 +237,23 @@ WASM_API_EXTERN wasmtime_error_t* wasmtime_linker_instantiate(
     wasm_trap_t **trap
 );
 
+/**
+ * \brief Instantiates a #wasm_module_t with the items defined in this linker for an async store.
+ *
+ * This is the same as #wasmtime_linker_instantiate but used for async stores 
+ * (which requires functions are called asynchronously). The returning #wasmtime_call_future_t 
+ * must be polled using #wasmtime_call_future_poll, and is owned and must be deleted using #wasmtime_call_future_delete.
+ * The future's results are retrieved using `wasmtime_call_future_get_results after polling has returned true marking 
+ * the future as completed.
```

Comment:
This isn't quite right since the main result is transmitting through the `instance` pointer I think? Given my suggestion above though it would mean that all results are transmitted through provided pointers, too.

Reply:
Correct - I've updated these docs a bit.